src/Illuminate/Http/Client/Factory.php

```diff
@@ -191,6 +191,21 @@ public function assertSent($callback)
         );
     }
 
+    /**
+     * Assert that a specific request / response pair was recorded matching a given truth test.
+     *
+     * @param callable $callback
+     * @param int $index
+     * @return void
+     */
+    public function assertSentByIndex($callback, $index)
+    {
+        PHPUnit::assertTrue(
+            $callback($this->recorded[$index][0], $this->recorded[$index][1]),
+            'An expected request was not recorded.'
```

Comment:
This should mention the index, I think.

Reply:
Agreed. Do you have any grammar suggestions?

I was thinking 
```
"An expected request was not recorded at index {$index}."
```